cpp/oneapi/dal/algo/knn/test/serialization.cpp

```diff
@@ -129,11 +130,21 @@ class knn_serialization_test : public te::float_algo_fixture<std::tuple_element_
 
     void compare_infer_results(const infer_result<task_t>& actual,
                                const infer_result<task_t>& reference) {
-        INFO("compare responses") {
-            te::check_if_tables_equal<float_t>(actual.get_responses(), reference.get_responses());
+        if constexpr (is_classification) {
+            INFO("compare responses") {
+                te::check_if_tables_equal<float_t>(actual.get_responses(),
+                                                   reference.get_responses());
+            }
+        }
+        if constexpr (is_search) {
```

Comment:
or if result_options is knn::result_options::indices, knn::result_options::distances